File: Soultion/Closest-Prime-Numbers-in-Range.java
class Solution {
    public int[] closestPrimes(int left, int right) {

        boolean[] isPrime = new boolean[right + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;

        for (int i = 2; i * i <= right; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= right; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        List<Integer> primes = new ArrayList<>();
        for (int num = left; num <= right; num++) {
            if (isPrime[num]) {
                primes.add(num);
            }
        }

        if (primes.size() < 2) {
            return new int[] { -1, -1 };
        }

        int minGap = Integer.MAX_VALUE;
        int[] bestPair = new int[2];

        for (int i = 0; i < primes.size() - 1; i++) {
            int gap = primes.get(i + 1) - primes.get(i);
            if (gap < minGap) {
                minGap = gap;
                bestPair[0] = primes.get(i);
                bestPair[1] = primes.get(i + 1);
            }
        }

        return bestPair;
    }
}

File: Soultion/Closest-Prime-Numbers-in-Range.py
class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:
        primes = []
        
        for num in range(left, right + 1):
            if num < 2:
                continue
            if num in (2, 3):
                primes.append(num)
                continue  
            
            if num % 2 == 0 or num % 3 == 0:
                continue
            
            i = 5
            is_prime = True  
            while i * i <= num:
                if num % i == 0 or num % (i + 2) == 0:
                    is_prime = False  
                    break  
                i += 6 
            
            if is_prime:
                primes.append(num)
        
        if len(primes) < 2:
            return [-1, -1]

        min_gap = float("inf")
        best_pair = [-1, -1]
        
        for i in range(len(primes) - 1):
            gap = primes[i + 1] - primes[i]
            if gap < min_gap:
                min_gap = gap
                best_pair = [primes[i], primes[i + 1]]

        return best_pair  
